src/documents/Priorities.md

```diff
@@ -0,0 +1,107 @@
+---
+title: Priorities
+layout: default
+category: API
+---
+
+Many elements of an Infusion application (in particular, [Listeners](InfusionEventSystem.md#registering-a-listener-to-an-event), 
+[Model Listeners](ChangeApplierAPI.md#model-listener-declaration), [Options Distributions](IoCSS.md) and [Subcomponents](SubcomponentDeclaration.md) ) 
+allow the use of a standardised entry named ``priority`` which allows the position of that element within a list of such elements to be adjusted.
+Depending on what the element is, the effect of the priority will be different &#8212; for example, a listener may be notified earlier or later than another
+listener, or a set of options or grades may take priority over another during options merging.
+
+The priority system is designed to be "open" in the face of evolving application designs that are worked on by different groups. As well as standard
+fixed priorities (represented by numbers or _extremal values_ such as `"first"` or `"last"`), Infusion priorities may be _constraint-based_ in which
+an element defines its priority only with respect to another element which is identified by _namespace_. These constraint-based priorities are
+much less brittle than the use of fixed priorities, and are recommended whenever a priority directive seems to be required.
+
+## Supported values for priorities ##
+
+
+<table>
+  <thead>
+    <tr>
+      <td>Type</td>
+      <td>Description</td>
+      <td>Examples</td>
+    </tr>
+  </thead>
+<tr><td>Numeric</td>
+<td>An <em>integer value</em> representing a fixed priority (not generally recommended). In general, higher numbers represent "greater priority" with whatever meaning
+that takes in the relevant context. For example, a listener with higher priority will fire before one with a lower priority, or options with higher priority 
+will merge on top of options with a lower priority</td><td>1, 10, 100</td></tr>
+
+<tr><td>Constraint</td>
+<td>Either of the strings <code>before</code> or <code>after</code> followed by the <code>namespace</code> of some other element (the "target element") of the same type, separated by a colon.
+The framework will sort all the elements in the same set so that this element will be placed immediately before or after the target element, 
+unless a further constraint positions a third listener in between them.</td><td><code>"before:bindMarkup"</code>, <code>"after:computeLayout"</code></td></tr>
+
+<tr><td>Extremal Priority</td>
+<td>Either of the strings <code>first</code> or <code>last</code>, either by themselves, or followed by an <strong>extremal priority class</strong> separated by a colon. Elements annotated <code>first</code> or
```

Comment:
The two possible values for an "extremal priority class" should be more clearly set off and stated earlier. Right now, they're buried at the end of a long narrative, leaving the reader wondering "What is this class? Where is it defined? Do I make it up, or does the framework offer it?" etc...